### ISSUE TITLE
[Masthead]: QA: Screen reader identifies the menu options in tab/shift tab and up/down arrow navigation even after collapsing the menu

### ISSUE THREAD

**Issue by Gopi916** [carbon-design-system/carbon-for-ibm-dotcom#8852]
### Description

Page Element:
User profile menu

Issue:
Screen reader identifies the menu options in tab/shift tab and up/down arrow navigation even after collapsing the menu

Expected:
After collapsing the menu the options should be hidden from the screen reader

User Impact:
If the screen reader identifies the visually hidden content, it tends to mislead users who are blind. 

### Component(s) impacted

Masthead
Default
With custom type ahead
Search open on load
With L1
With alternate logo and tool tip

### Browser

Firefox

### Carbon for IBM.com version

v1.34.0-rc.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://www.ibm.com/standards/carbon/web-components/?path=/story/

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
